Makefile

```diff
@@ -65,12 +65,15 @@ deploy:
 	kustomize build config/default | kubectl apply -f -
 
 protoc-docker:
-	docker build - -t protoc < hack/grpc-protoc.Dockerfile
+	docker build --quiet - -t protoc < hack/grpc-protoc.Dockerfile
 
 protobuf: protoc-docker
-	docker run -v `pwd`:/eco -w /eco protoc:latest -I=api/proxy --go_out=plugins=grpc:api/proxy api/proxy/proxy.proto
+	docker run -v `pwd`:/eco -w /eco protoc:latest -I=api/proxy --go_out=plugins=grpc:api/proxy api/proxy/v1/proxy.proto
```

Comment:
By far the biggest change is not a lintier but the v1 here. Why a version directory? Do we anticipate supporting multiple versions?

Reply:
I didn't give it much thought. I just ran the tool with it's default checks. And this was one of the lint failures, probably inspired by this Uber protobuf style guide: https://github.com/uber/prototool/blob/dev/style/README.md#package-versioning

I guess it may be important. If in future we separate out the proxy and / or if we want to change the protocol, perhaps there'd be a need to have the  proxy  support multiple protocol versions during an upgrade of the etcd-cluster-operator.
That's hand waving, but perhaps versioning the API from the start would be good practice.

And if you version the API,  `buf` or `prototool` can check that you don't make breaking  changes to the API.

The reason I made this branch is that I was looking trying to figure out why the new `protoc` Docker image wasn't running on my laptop. So I went searching for other, pre-built protoc Docker images and came across https://github.com/uber/prototool and then https://github.com/bufbuild/buf.

It turns out that neither of these are meant for generating the protobuf code, but they seemed like interesting tools so I tried them.

And the reason I couldn't run make verify on my laptop was to do with my Fedora Docker configuration.

So I'm happy to close this PR unless you think there is merit in running these checks.

Reply:
> […] perhaps there'd be a need to have the proxy support multiple protocol versions during an upgrade of the etcd-cluster-operator.

I also thought down this path. In theory we'll deploy everything in lock-step. But there is the upgrade case where a backup might be running while we do an upgrade, which should still work.

> […] but perhaps versioning the API from the start would be good practice.

It's a reasonable argument. I think I was just surprised upon seeing a `v1` get introduced as part of a linting change.

> […] they seemed like interesting tools so I tried them.

I'm not familiar with either, but happy to give them a try. We can always remove them if we find them difficult to work with.

> So I'm happy to close this PR unless you think there is merit in running these checks.

I'm happy to merge this. You're right that the v1 is a good practice to get into even if we don't *think* we'll need it now. We can always remove the linting later if it becomes a problem, and in general I like having linting in CI.